### DISCUSSION TITLE
TS Error with your on-site example library '@internationalized/date' in 3.5.5 version.

### DISCUSSION THREAD

**Discussion by uigywnkiub** [heroui-inc/heroui#3536] [Feedback]
The error - **Type 'CalendarDate' is not assignable to type 'DateValue'.
  Type 'CalendarDate' is not assignable to type 'ZonedDateTime'.
    Property '#private' in type 'CalendarDate' refers to a different member that cannot be accessed from within type 'ZonedDateTime'.ts(2322)**
On 3.5.4 all was good.
```js
const toCalendarDate = (date: Date) => {
  return new CalendarDate(getYear(date), getMonth(date) + 1, getDate(date))
}

// startOfMonth() and startOfToday() functions are from 'date-fns'

 const [selectedDate, setSelectedDate] = useState<RangeValue<DateValue>>({
    start: toCalendarDate(startOfMonth(startOfToday())),
    end: toCalendarDate(endOfMonth(endOfToday())),
  })
```

**Comment by lorenzo-dallamuta** [comment]
I see the same thing on 3.5.6

This is especially frustrating because it happens when using the Calendar component from React Aria Components as a starting point.

Specifically when having `CalendarContext` interact with `useCalendarState`.

```
[props, ref] = useContextProps(props, ref, CalendarContext);
let state = useCalendarState({props);
```

EDIT:
at least in my case, it was a wrong peer dependency version that was lingering in the node_modules directory, I deleted that reinstalled and solved my issue 😅

**Comment by pavitra-infocusp** [comment]
I can reproduce this same error with `@internationalized/date` 3.6.0 on my project.

```
Argument of type '{ start: CalendarDate; end: CalendarDate; }' is not assignable to parameter of type 'RangeValue<CalendarDate>'.
  Types of property 'start' are incompatible.
    Type 'import("/home/pavitra/Projects/turnstile-fe/node_modules/.pnpm/@internationalized+date@3.6.0/node_modules/@internationalized/date/dist/types").CalendarDate' is not assignable to type 'import("/home/pavitra/Projects/turnstile-fe/node_modules/.pnpm/@internationalized+date@3.5.5/node_modules/@internationalized/date/dist/types").CalendarDate'.
      Property '#private' in type 'CalendarDate' refers to a different member that cannot be accessed from within type 'CalendarDate'.ts(2345)
```

**Comment by wingkwong** [comment]
see https://github.com/nextui-org/nextui/issues/4154#issuecomment-2505243155
